src/scirpy/ir_dist/metrics.py

```diff
@@ -392,6 +394,698 @@ def _compute_block(self, seqs1, seqs2, origin):
         return result
 
 
+class TCRdistDistanceCalculator:
+    """Computes pairwise distances between TCR CDR3 sequences based on the "tcrdist" distance metric.
+
+    The code of this class is heavily based on: https://github.com/agartland/pwseqdist/blob/master/pwseqdist
+
+    Using default weight, gap penalty, ntrim and ctrim is equivalent to the
+    original distance published in Dash et al, (2017).
+
+    Parameters
+    ----------
+    dist_weight : int
```

Comment:
Please specify types as type hints in the function, and omit them in the documentation. Sphinx will automatically retrieve the types from the type hints and show them in the docs website.